### PR TITLE
ThreadLocalSingleWebDriverPool improvements

### PR DESCRIPTION
**Pull request by emaks**

### CODE REVIEW

**Comment by barancev** [comment]
What's the essense of this improvement? Could you please add a sample/a test case that shows the benefits.

**Comment by emaks** [comment]
changes in getDriver() - decrease code complexity
``` try { } catch {} ``` block for ``` driver.quit(); ``` - skip errors related to problems with closing browser
changing type for ``` driverToKeyMap  ```(``` HashMap ``` to ``` ConcurrentHashMap ```) - problems related to the case when different threads are adding data at the same time

File: src/main/java/ru/stqa/selenium/factory/ThreadLocalSingleWebDriverPool.java [hunk 1]
```diff
@@ -39,30 +40,24 @@ public void run() {
...
   @Override
   public WebDriver getDriver(String hub, Capabilities capabilities) {
-    String newKey = createKey(capabilities, hub);
     if (tlDriver.get() == null) {
+      return createNewDriver(capabilities, hub);
+    }
+    String key = driverToKeyMap.get(tlDriver.get());
+    if (key == null) {
+      // The driver was dismissed in a wrong way
+      return createNewDriver(capabilities, hub);
```
Comment (selenium-course) [inline src/main/java/ru/stqa/selenium/factory/ThreadLocalSingleWebDriverPool.java]: I think that WebDriver return in createNewDriver() method makes code less readable and obvoius. Four (4!) return statements let the code smell.

**Comment by barancev** [comment]
Commits 88ca8f173cc4eed42a1a9fc5e2dda229c890b0f3 and b873f908e986314af3d52841487011b6962d5636 cover improvements suggested by this PR.

Multiple return statements fit my code style :)

**Comment by emaks** [comment]
@barancev ok :) How about this variant?
```
  public synchronized WebDriver getDriver(URL hub, Capabilities capabilities) {
    dismissDriversInFinishedThreads();
    String newKey = createKey(capabilities, hub);
    String key = driverToKeyMap.get(tlDriver.get());
    if (tlDriver.get() == null) {
      createNewDriver(capabilities, hub);
    } else if (key == null) { // The driver was dismissed
      createNewDriver(capabilities, hub);
    } else if (!newKey.equals(key)) { // A different flavour of WebDriver is required
      dismissDriver(tlDriver.get());
      createNewDriver(capabilities, hub);
    } else if (!alivenessChecker.isAlive(tlDriver.get())) { // Check the browser is alive
      dismissDriver(tlDriver.get());
      createNewDriver(capabilities, hub);
    }
    return tlDriver.get();
  }

**Comment by barancev** [comment]
I prefer "return as soon as possible" style.

**Pull request closed without merging**